### PR TITLE
Update duende versions to 6.0.4

### PR DESCRIPTION
**Pull request by HaoK**
For https://github.com/dotnet/aspnetcore/issues/41315

### CODE REVIEW

File: eng/Versions.props [hunk 1]
```diff
@@ -245,11 +245,11 @@
     <GrpcAuthVersion>2.43.0</GrpcAuthVersion>
     <GrpcNetClientVersion>2.43.0</GrpcNetClientVersion>
     <GrpcToolsVersion>2.43.0</GrpcToolsVersion>
-    <DuendeIdentityServerAspNetIdentityVersion>5.2.0</DuendeIdentityServerAspNetIdentityVersion>
-    <DuendeIdentityServerEntityFrameworkVersion>5.2.0</DuendeIdentityServerEntityFrameworkVersion>
-    <DuendeIdentityServerVersion>5.2.0</DuendeIdentityServerVersion>
-    <DuendeIdentityServerStorageVersion>5.2.0</DuendeIdentityServerStorageVersion>
-    <DuendeIdentityServerEntityFrameworkStorageVersion>5.2.0</DuendeIdentityServerEntityFrameworkStorageVersion>
+    <DuendeIdentityServerAspNetIdentityVersion>6.0.4</DuendeIdentityServerAspNetIdentityVersion>
```
Comment (premun) [inline eng/Versions.props:248]: I think this variable is not used anywhere nor I was able to find a package with that name

**Comment by brockallen** [comment]
We have a 6.1 that will be released in the next week or so, FWIW.

File: eng/Versions.props (hunk 1 above)
Comment (HaoK) [reply eng/Versions.props:248]: The package should be: https://www.nuget.org/packages/Duende.IdentityServer.AspNetIdentity/6.0.4 do you see that?

Comment (premun) [reply eng/Versions.props:248]: I can see the NuGet but this property is not used anywhere in the repo so maybe it's not needed?

Comment (HaoK) [reply eng/Versions.props:248]: Ok I'll try a build

Comment (HaoK) [reply eng/Versions.props:248]: Definitely need this package:

```
src/Identity/ApiAuthorization.IdentityServer/src/Microsoft.AspNetCore.ApiAuthorization.IdentityServer.csproj(0,0): error NU1102: (NETCORE_ENGINEERING_TELEMETRY=Restore) Unable to find package Duende.IdentityServer.AspNetIdentity with version (>= 6.0.4)
  - Found 20 version(s) in dotnet-public [ Nearest version: 5.2.1 ]
```

Comment (premun) [reply eng/Versions.props:248]: Oh sorry, it didn't get added. It's there now!

Comment (HaoK) [reply eng/Versions.props:248]: Thanks, looks like we also need Automapper 11 as a result of these updated packages too...

https://www.nuget.org/packages/AutoMapper/

**Comment by brockallen** [comment]
@HaoK Here's the preview if you wanted to test sooner: https://www.nuget.org/packages/Duende.IdentityServer/6.1.0-preview.3

File: eng/Versions.props (hunk 1 above)
Comment (premun) [reply eng/Versions.props:248]: Done

Comment (HaoK) [reply eng/Versions.props:248]: Another updated package: IdentityModel 6.0.0  https://www.nuget.org/packages/IdentityModel/

File: src/Identity/ApiAuthorization.IdentityServer/src/Extensions/DefaultClientRequestParametersProvider.cs [hunk 2]
```diff
@@ -12,20 +12,24 @@ internal sealed class DefaultClientRequestParametersProvider : IClientRequestPar
...
 
     public IOptions<ApiAuthorizationOptions> Options { get; }
 
+    public IIssuerNameService NameService { get; }
+
     public IDictionary<string, string> GetClientParameters(HttpContext context, string clientId)
     {
         var client = Options.Value.Clients[clientId];
-        var authority = context.GetIdentityServerIssuerUri();
+        var authority = NameService.GetCurrentAsync().GetAwaiter().GetResult();
```
Comment (HaoK) [inline src/Identity/ApiAuthorization.IdentityServer/src/Extensions/DefaultClientRequestParametersProvider.cs:32]: This is a little gross but the old obsolete API is sync, and IIssuerNameService is async only

**Review by wtgodbe — APPROVED** [review]

File: src/Identity/ApiAuthorization.IdentityServer/src/Extensions/DefaultClientRequestParametersProvider.cs (hunk 2 above)
Comment (sebastienros) [reply src/Identity/ApiAuthorization.IdentityServer/src/Extensions/DefaultClientRequestParametersProvider.cs:32]: At what point can we make breaking changes and make this one async? Isn't a version like 7.0 a good opportunity? Or at least add/use the async one and make the `GetClientParameters` obsolete.

**Comment by brockallen** [comment]
FWIW https://www.nuget.org/packages/Duende.IdentityServer/6.1.0

:)

**Comment by HaoK** [comment]
Cool, we'll update to 6.1 for preview6

**Pull request merged**